components/server/src/workspace/gitpod-server-impl.ts

```diff
@@ -502,12 +502,21 @@ export class GitpodServerImpl implements GitpodServerWithTracing, Disposable {
         if (workspace.deleted) {
             throw new ResponseError(ErrorCodes.PERMISSION_DENIED, "Cannot (re-)start a deleted workspace.");
         }
-        const envVars = this.userDB.getEnvVars(user.id);
+        const userEnvVars = this.userDB.getEnvVars(user.id);
+        let projectEnvVarsPromise: Promise<ProjectEnvVar[]> = Promise.resolve([]);
+        if (workspace.projectId) {
+            try {
+                await this.guardProjectOperation(user, workspace.projectId, "get");
+                projectEnvVarsPromise = this.projectsService.getProjectEnvironmentVariables(workspace.projectId);
+            } catch (error) {
+                log.warn({ userId: user.id }, "User doesn't have access to the Project, thus not loading Project environment variables:", error);
```

Comment:
Is this really `warn`? IMO it should be `debug`: this might very well happen, if 1.) we have a prominten, public repo 2.) there is a Gitpod team for it only very few people are part of.

Reply:
Good catch, 100% agree. I wanted to make this more visible, as we're still working on properly segregating "public access" vs "Team/Project access" for public repos, but in this location the segregation is done properly so there is probably no need to be loud.